packages/core/contracts/cross-chain-oracle/chain-adapters/Arbitrum_ParentMessenger.sol

```diff
@@ -0,0 +1,153 @@
+        _sendMessageToChild(dataSentToChild, childMessenger);
+    }
+
+    /**
+     * @notice Changes the address of the parent messenger on L2 via the child messenger.
+     * @dev The caller of this function must be the owner. This should be set to the DVM governor.
+     * @param newParentMessenger the new parent messenger contract to be set on L2.
+     */
+    function setChildParentMessenger(address newParentMessenger) public payable onlyOwner {
+        bytes memory dataSentToChild = abi.encodeWithSignature("setParentMessenger(address)", newParentMessenger);
+        _sendMessageToChild(dataSentToChild, childMessenger);
+    }
+
+    /**
+     * @notice Sends a message to the child messenger via the canonical message bridge.
+     * @dev The caller must be the either the OracleHub or the GovernorHub. This is to send either a
+     * price or initiate a governance action to the OracleSpoke or GovernorSpoke on the child network.
+     * @dev The recipient of this message is the child messenger. The messenger must implement processMessageFromParent
+     * which then forwards the data to the target either the OracleSpoke or the governorSpoke depending on the caller.
+     * @param data data message sent to the child messenger. Should be an encoded function call or packed data.
+     */
+    function sendMessageToChild(bytes memory data) public payable override onlyHubContract() nonReentrant() {
+        address target = msg.sender == oracleHub ? oracleSpoke : governorSpoke;
+        bytes memory dataSentToChild =
+            abi.encodeWithSignature("processMessageFromCrossChainParent(bytes,address)", data, target);
+        _sendMessageToChild(dataSentToChild, target);
+    }
+
+    /**
+     * @notice Process a received message from the child messenger via the canonical message bridge.
+     * @dev The caller must be the the child messenger, sent over the canonical message bridge.
+     * @dev Note that only the OracleHub can receive messages from the child messenger. Therefore we can always forward
+     * these messages to this contract. The OracleHub must implement processMessageFromChild to handle this message.
+     * @param data data message sent from the child messenger. Should be an encoded function call or packed data.
+     */
+    function processMessageFromCrossChainChild(bytes memory data)
+        public
+        onlyFromCrossDomainAccount(childMessenger)
+        nonReentrant()
+    {
+        ParentMessengerConsumerInterface(oracleHub).processMessageFromChild(childChainId, data);
+        emit MessageReceivedFromChild(data, childMessenger, oracleHub);
+    }
+
+    function _sendMessageToChild(bytes memory data, address target) internal {
+        uint256 seqNumber =
+            sendTxToL2NoAliassing(
+                childMessenger,
+                owner(), // This is the address that will send ETH refunds for any failed messages.
```

Comment:
owner() will be the owner of the `OracleBase`, right? so this, as it's the `ParentMessenger`, will be the `Governor` address, but on L2? this might be suboptimal as we won't be able to re-claim these funds, no?

Reply:
I changed the ParentMessenger so that the owner can set a refund address